scripts/build_android_library.sh

```diff
@@ -50,6 +50,7 @@ build_android_native_library() {
     -DEXECUTORCH_BUILD_EXTENSION_MODULE=ON \
     -DEXECUTORCH_BUILD_EXTENSION_RUNNER_UTIL=ON \
     -DEXECUTORCH_BUILD_EXTENSION_TENSOR=ON \
+    -DEXECUTORCH_BUILD_EXTENSION_TRAINING=ON \
```

Comment:
So why are we turning this on by default now?

Reply:
@kirklandsign  and @larryliu0820  recommended just piggybacking on top of the existing android ci to get build coverage.